homeassistant/components/tod/binary_sensor.py

```diff
@@ -226,6 +226,29 @@ def _calculate_boundary_time(self) -> None:
         self._time_after += self._after_offset
         self._time_before += self._before_offset
 
+    def _add_one_dst_aware_day(self, a_date: datetime) -> datetime:
```

Comment:
Can you use `find_next_time_expression_time` here?

https://github.com/home-assistant/core/blob/dev/homeassistant/util/dt.py#L345

Reply:
Thanks Nick.  I updated the code to use `find_next_time_expression_time`.  In addition to simplifying the code, it also makes the ToD sensor better honor the start/end time during DST transitions.    There's a degenerate case now where when a sensor is scheduled to start/end _during_ an non-existent hour, the start/end time gets pushed to the next available time (thanks to `find_next_time_expression_time`) essentially making the ToD sensor to not activate.  I thinks it's fine (please chime in if not) but we probably want to add a note in the documentation about that.  For use cases where a sensor should absolutely run for a certain duration, we could enhance the ToD integration (in the future) to support that.  (optionally make it start time + duration based instead of start/end time based).